### ISSUE TITLE
How many epochs should I train?The default is 20. Is that too many?

### ISSUE THREAD

**Issue by cizhenshi** [harsh-99/SCL#6]

**Comment by harsh-99** [comment]
Hi, 
The epoch by default is 20. However it depends on your wish how much you want. By my experience you can get good results in first 10 epochs.

**Comment by cizhenshi** [comment]
In [https://github.com/harsh-99/SCL/blob/master/lib/model/faster_rcnn/vgg16_dfrcnn.py](url)
`class netD_img(nn.Module):
  def __init__(self, beta=1, ch_in=1024, ch_out=1024, W=38, H=75, stride_1=1, padding_1=1, kernel=3):
    super(netD_img, self).__init__()
    self.conv_image = nn.Conv2d(ch_in, ch_out, stride=stride_1, padding=padding_1, kernel_size=kernel)
    self.bn_image = nn.BatchNorm2d(ch_out)
    self.fc_1_image = nn.Linear(1, 2)
    self.ch_out = ch_out
    self.relu = nn.ReLU(inplace=True)
    self.maxpool = nn.MaxPool2d(kernel_size=2)
    self.bn_2 = nn.BatchNorm2d(ch_out)
    #self.softmax = nn.Softmax()
    #self.logsoftmax = nn.LogSoftmax()

  def forward(self, x):
    x = self.conv_image(x)
    x = self.relu(x)
    x = self.bn_image(x)
    x = self.maxpool(x)
    x = self.bn_2(x)
    # convert to 1024*W*H x 1.
    x = flatten(x)
    x = torch.transpose(x, 0, 1)
    x = self.fc_1_image(x)
    # 1 x n vector
    #y = self.softmax(x)
    #x = self.logsoftmax(x)
    #return x, y
    return x`

I noticed that you used the BN layer and max pool layer, which didn't seem to exist in the original implementation.  Are they necessary and why add these layers?
